.github/workflows/infrastructure.yml

```diff
@@ -0,0 +1,173 @@
+
+    - name: Timezone 🌐
+      run: echo "Timezone set to ${{ env.timezone }}"
+
+# 2. Base infrastructure creation
+  base:
+    name: Base 🧱
+    needs: setup
+    environment: ${{ needs.setup.outputs.environment }}
+    runs-on: [ self-hosted, APIM, infrastructure ]
+    steps:
+
+    - name: Pre-production 💫
+      if: contains('["dev", "staging"]', env.target)
+      run: echo "type=Preproduction" >> $GITHUB_ENV
+
+    - name: Production 💫
+      if: ${{ 'prod' == env.target }}
+      run: echo "type=Production" >> $GITHUB_ENV
+
+    - name: Tags 🏷️
+      run: echo tags='Environment=${{ env.type }}' \
+        'Product=${{ env.product }}' \
+        'Region=${{ vars.REGION }}' \
+        'Team=development' >> $GITHUB_ENV
+
+    - name: Login 🔐
+      uses: azure/login@v1
+      with:
+        creds: ${{ secrets.AZURE_CREDENTIALS }}
+
+    - name: Azure defaults ✨
+      uses: azure/cli@v1.0.0
+      with:
+        inlineScript: |
+          # Basic
+          az configure --defaults location=${{ vars.REGION }}
+          az configure --defaults group=rg-${{ env.product }}-${{ env.target }}-001
+
+          # AZ CLI extensions upgrade
+          az extension add --name containerapp --upgrade
+
+    - name: Resource group 🏗️
+      uses: azure/cli@v1.0.0
+      with:
+        inlineScript: |
+          az group create \
+          --name rg-${{ env.product }}-${{ env.target }}-001 \
+          --tags ${{ env.tags }}
+
+    
+    - name: Container registry 📦️
+      uses: azure/cli@v1.0.0
+      with:
+        inlineScript: |
+          az acr create \
+          --name cr${{ env.product }}${{ env.target }}${{ vars.REGION }}001 \
+          --sku Standard \
+          --admin-enabled true \
+          --tags ${{ env.tags }}
+
+    - name: Container app environment 🗃️
+      uses: azure/cli@v1.0.0
+      with:
+        inlineScript: |
+          az containerapp env create \
+          --name cae-${{ env.product }}-${{ env.target }}-${{ vars.REGION }}-001 \
+          --tags ${{ env.tags }}
+
+    
+    - name: Container app - MDM 📄
+      uses: azure/cli@v1.0.0
+      with:
+        inlineScript: |
+          az containerapp create \
+          --name ca-mdm-${{ env.product }}-${{ env.target }}-${{ vars.REGION }}-001 \
+          --environment cae-${{ env.product }}-${{ env.target }}-${{ vars.REGION }}-001 \
+          --registry-server cr${{ env.product }}${{ env.target }}${{ vars.REGION }}001.azurecr.io \
+          --registry-username $(az acr credential show -n cr${{ env.product }}${{ env.target }}${{ vars.REGION }}001 --query username | tr -d '"') \
+          --cpu 0.25 \
+          --memory 0.5Gi \
+          --min-replicas 1 \
+          --ingress internal \
+          --target-port ${{ vars.MDM_PORT }} \
+          --revision-suffix ${{ vars.REVISION_SUFFIX }} \
+          --revisions-mode multiple \
+          --transport http2 \
+          --tags ${{ env.tags }}
+
+    - name: Container app - TFS 📄
```

Comment:
Just a note, this would generate the TFS Container App from MDM repository. Shouldn't this be split and added to TFS repo?

Reply:
Yes they will be split, this is just to test multiple container deployment.